fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -223,23 +149,73 @@
                   @Nonnull IndexStatePrecondition indexStatePrecondition,
                   boolean useSynchronizedSession,
                   long leaseLengthMillis,
-                  boolean trackProgress) {
+                  boolean trackProgress,
+                  @Nonnull IndexFromIndexPolicy indexFromIndex) {
         this.runner = runner;
-        this.recordStoreBuilder = recordStoreBuilder;
         this.index = index;
-        this.recordTypes = recordTypes;
-        this.configLoader = configLoader;
-        this.config = config;
-        this.limit = config.maxLimit;
-        this.syntheticIndex = syntheticIndex;
-        this.indexStatePrecondition = indexStatePrecondition;
-        this.useSynchronizedSession = useSynchronizedSession;
-        this.leaseLengthMills = leaseLengthMillis;
-        this.trackProgress = trackProgress;
-
-        this.recordsRange = computeRecordsRange();
-        timeOfLastProgressLogMillis = System.currentTimeMillis();
-        totalRecordsScanned = new AtomicLong(0);
+        this.indexFromIndexPolicy = indexFromIndex;
+
+        this.common = new IndexingCommon(runner, recordStoreBuilder,
+                index, recordTypes, configLoader, config,
+                syntheticIndex,
+                indexStatePrecondition,
+                trackProgress,
+                useSynchronizedSession,
+                leaseLengthMillis
+            );
+    }
+
+    @Nonnull
+    private IndexingByIndex getScannerByIndex() {
```

Comment:
The"scanner" wording could be confusing now, perhaps replacing "scanner" with "underlying" would be helpful. I'm not sure though.

Reply:
I was looking for something that will describe the processes of scanning the records (this way or another) and indexing them. "underlying" doesn't seem descriptive enough (unless it has a sub meaning that I - as a non native English speaker - am missing). Would replacing "scanner" with "compiler" be too confusing?

Reply:
How about indexer? 
`getIndexer()` 
`getIndexerByRecord()` 
`getIndexerByIndex()` 
Is that less confusing?

Reply:
I think Indexer is less confusing, I suppose